sklearn/tests/test_pipeline.py

```diff
@@ -266,6 +266,31 @@ def test_pipeline_sample_weight_unsupported():
     )
 
 
+class Dummy(object):
+    def __init__(self):
+        pass
+
+    def cache(self, func):
+        return func
+
+    cachedir = None
+
+
+class Wrong_Dummy(object):
+    def __init__(self):
+        pass
+
+
+def test_pipeline_with_cache_attribute():
+    X = np.array([[1, 2]])
+    pipe = Pipeline([('transf', Transf()), ('clf', Mult())], memory=Dummy())
+    pipe.fit(X, y=None)
+
+    pipe = Pipeline([('transf', Transf()), ('clf', Mult())],
+                    memory=Wrong_Dummy())
+    assert_raises(ValueError, pipe.fit, X)
```

Comment:
Use assert_raises_message or assert_raises_regex to check the message which is raised.